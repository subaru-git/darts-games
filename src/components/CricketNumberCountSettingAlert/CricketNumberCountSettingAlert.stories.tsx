/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import CricketNumberCountSettingAlert from './CricketNumberCountSettingAlert'

export default {
  title: 'Components/CricketNumberCountSettingAlert',
  component: CricketNumberCountSettingAlert,
} as ComponentMeta<typeof CricketNumberCountSettingAlert>

const Template: ComponentStory<typeof CricketNumberCountSettingAlert> = (args) => (
  <CricketNumberCountSettingAlert {...args} />
)

export const Default = Template.bind({})
Default.args = {
  isOpen: true,
}
