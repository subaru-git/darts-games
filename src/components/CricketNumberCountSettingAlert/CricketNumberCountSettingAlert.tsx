import React, { FC, useRef } from 'react'
import {
  AlertDialog,
  AlertDialogBody,
  AlertDialogCloseButton,
  AlertDialogContent,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogOverlay,
  Button,
} from '@chakra-ui/react'
import useLocale from '@/hooks/locale'

type CricketNumberCountSettingAlertProps = {
  isOpen: boolean
  onClose: () => void
  onNewGame: () => void
}

const CricketNumberCountSettingAlert: FC<CricketNumberCountSettingAlertProps> = ({
  isOpen,
  onClose,
  onNewGame,
}) => {
  const cancelRef = useRef<HTMLButtonElement>(null)
  const { t } = useLocale()
  return (
    <AlertDialog
      isOpen={isOpen}
      onClose={onClose}
      leastDestructiveRef={cancelRef}
      blockScrollOnMount={false}
    >
      <AlertDialogOverlay />
      <AlertDialogContent>
        <AlertDialogHeader fontSize='lg' fontWeight='bold'>
          Start New Game?
        </AlertDialogHeader>
        <AlertDialogCloseButton />
        <AlertDialogBody>{t.CRICKET_NUMBER_COUNT_START_NEW_GAME_WARNING}</AlertDialogBody>
        <AlertDialogFooter>
          <Button
            colorScheme='blue'
            ml={3}
            onClick={() => {
              onNewGame()
              onClose()
            }}
          >
            OK
          </Button>
          <Button onClick={onClose} ref={cancelRef} ml={3}>
            Cancel
          </Button>
        </AlertDialogFooter>
      </AlertDialogContent>
    </AlertDialog>
  )
}
export default CricketNumberCountSettingAlert
