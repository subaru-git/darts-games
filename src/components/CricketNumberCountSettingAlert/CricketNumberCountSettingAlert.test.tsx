import { fireEvent, render, screen } from '@testing-library/react'
import CricketNumberCountSettingAlert from './CricketNumberCountSettingAlert'

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }))

test('should rendering', () => {
  const onCloseMock = jest.fn()
  const onNewGameMock = jest.fn()
  const { container } = render(
    <CricketNumberCountSettingAlert
      isOpen={true}
      onClose={onCloseMock}
      onNewGame={onNewGameMock}
    />,
  )
  fireEvent.click(screen.getByText('OK'))
  expect(onNewGameMock).toBeCalledTimes(1)
  expect(onCloseMock).toBeCalledTimes(1)
  expect(container).toMatchSnapshot()
})
