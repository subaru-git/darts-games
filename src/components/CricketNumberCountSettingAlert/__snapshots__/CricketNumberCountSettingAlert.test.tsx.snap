// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
